Tests/FoundationEssentialsTests/DateIntervalTests.swift

```diff
@@ -50,6 +50,105 @@ final class DateIntervalTests : XCTestCase {
+        XCTAssertEqual(d3?.start, testInterval3.end)
+        XCTAssertEqual(d4?.duration, 1)
+        XCTAssertEqual(d4?.start, testInterval3.end)
+        
+        // Tests for non-existing intervals between subjects.
+        // |testInterval5|
+        //      |testInterval6|
+        //
+        // As a single timeline: |555|565656|666|
+        let testInterval5 = DateInterval(start: Date(timeIntervalSince1970: 0), end: Date(timeIntervalSinceReferenceDate: 0))
+        let testInterval6 = DateInterval(start: Date(timeIntervalSinceReferenceDate: -1), end: Date(timeIntervalSinceReferenceDate: 100))
+        let t5 = testInterval5.timeInterval(to: testInterval6)
+        XCTAssertEqual(t5, nil)
+        
+        let t6 = testInterval6.timeInterval(to: testInterval5)
+        XCTAssertEqual(t6, nil)
+        
+        let d5 = testInterval5.dateInterval(to: testInterval6)
+        XCTAssertEqual(d5, nil)
+        
+        let d6 = testInterval6.dateInterval(to: testInterval5)
+        XCTAssertEqual(d6, nil)
+        
+        // |---testInterval7---|
+        //    |testInterval8|
+        //
+        // As a single timeline: |777|787878|777|
+        let testInterval7 = DateInterval(start: Date(timeIntervalSince1970: 0), end: Date(timeIntervalSinceReferenceDate: 0))
+        let testInterval8 = DateInterval(start: Date(timeIntervalSince1970: 10), end: Date(timeIntervalSince1970: 20))
+        let t7 = testInterval7.timeInterval(to: testInterval8)
+        XCTAssertEqual(t7, nil)
+        
+        let t8 = testInterval8.timeInterval(to: testInterval7)
+        XCTAssertEqual(t8, nil)
+        
+        let d7 = testInterval7.dateInterval(to: testInterval8)
+        XCTAssertEqual(d7, nil)
+        
+        let d8 = testInterval8.dateInterval(to: testInterval7)
+        XCTAssertEqual(d8, nil)
+        
+        // |testInterval9|
+        // |testInterval10---|
+        let testInterval9 = DateInterval(start: Date(timeIntervalSince1970: 0), end: Date(timeIntervalSinceReferenceDate: 0))
+        let testInterval10 = DateInterval(start: Date(timeIntervalSince1970: 0), end: Date(timeIntervalSinceReferenceDate: 100))
+        let t9 = testInterval9.timeInterval(to: testInterval10)
+        XCTAssertEqual(t9, nil)
+        
+        let t10 = testInterval10.timeInterval(to: testInterval9)
+        XCTAssertEqual(t10, nil)
+        
+        let d9 = testInterval9.dateInterval(to: testInterval10)
+        XCTAssertEqual(d9, nil)
+        
+        let d10 = testInterval10.dateInterval(to: testInterval9)
+        XCTAssertEqual(d10, nil)
+        
+        // |testInterval11| on itself
+        let testInterval11 = DateInterval(start: Date(timeIntervalSince1970: 0), end: Date(timeIntervalSinceReferenceDate: 0))
+        let t11 = testInterval11.timeInterval(to: testInterval11)
+        XCTAssertEqual(t11, nil)
+        
+        let d11 = testInterval11.dateInterval(to: testInterval11)
+        XCTAssertEqual(d11, nil)
+    }
```

Comment:
I would have broken these down into separate test functions (one expecting non-nil and another nil intervals) but I implemented it like this to follow the lead of pre-existing test functions.